src/main/java/com/example/turboaz/service/TransactionService.java

```diff
@@ -0,0 +1,14 @@
+package com.example.turboaz.service;
+
+import com.example.turboaz.dtos.TransactionDto;
+import com.example.turboaz.enums.TransactionPurpose;
+import com.example.turboaz.enums.TransactionType;
+import com.example.turboaz.models.Transaction;
+
+import java.util.List;
+
+public interface TransactionService {
+    TransactionDto createTransaction(long listingId, TransactionPurpose purpose, TransactionType type, double amount);
+    List<Transaction> getAllTransactions();
```

Comment:
getAllTransactions must return the user's own transactions. that's why method must have parameter named userId.
Like this : getAllTransactions(int userId)